src/braket/aws/aws_device.py

```diff
@@ -397,60 +402,55 @@ def get_devices(
-                # Simulators are only instantiated in the same region as the AWS session
-                region_device_types = sorted(
-                    types if region == session_region else types - {AwsDeviceType.SIMULATOR}
-                )
-                region_device_arns = [
-                    result["deviceArn"]
-                    for result in session_for_region.search_devices(
-                        arns=arns,
-                        names=names,
-                        types=region_device_types,
-                        statuses=statuses,
-                        provider_names=provider_names,
-                    )
-                ]
-                device_map.update(
-                    {
-                        arn: AwsDevice(arn, session_for_region)
-                        for arn in region_device_arns
-                        if arn not in device_map
-                    }
+            session_for_region = AwsDevice._copy_aws_session(aws_session, region)
+            # Simulators are only instantiated in the same region as the AWS session
+            types_for_region = sorted(
+                types if region == session_region else types - {AwsDeviceType.SIMULATOR}
+            )
+            region_device_arns = [
+                result["deviceArn"]
+                for result in session_for_region.search_devices(
+                    arns=arns,
+                    names=names,
+                    types=types_for_region,
+                    statuses=statuses,
+                    provider_names=provider_names,
                 )
+            ]
+            device_map.update(
+                {
+                    arn: AwsDevice(arn, session_for_region)
+                    for arn in region_device_arns
+                    if arn not in device_map
+                }
+            )
         devices = list(device_map.values())
         devices.sort(key=lambda x: getattr(x, order_by))
         return devices
 
     @staticmethod
     def _get_devices_regions_set(
-        arns: Optional[List[str]],
-        provider_names: Optional[List[str]],
-    ) -> Set[str]:
+        types: Optional[Set[AwsDeviceType]], arns: Optional[List[str]], current_region: str
+    ) -> FrozenSet[str]:
         """Get the set of regions to call `SearchDevices` API given filters"""
-        device_regions_set = set(
-            AwsDevice.DEVICE_REGIONS[key][0] for key in AwsDevice.DEVICE_REGIONS
-        )
-        if provider_names:
-            provider_region_set = set()
-            for provider in provider_names:
-                for key in AwsDevice.DEVICE_REGIONS:
-                    if key in provider.lower():
-                        provider_region_set.add(AwsDevice.DEVICE_REGIONS[key][0])
-                        break
-            device_regions_set = device_regions_set.intersection(provider_region_set)
+        device_regions_set = set(AwsDevice.REGIONS)
```

Comment:
```
if types == {AwsDeviceType.SIMULATOR}:
    device_regions_set = {current_region}
else:
    device_regions_set = set(AwsDevice.REGIONS)
```

This has the added benefit of allowing bad region names in device_regions_set for simulator, which means it'll throw an error (as it should).